libgadget/fof.c

```diff
@@ -823,6 +858,8 @@ static void fof_reduce_groups(
     MPI_Type_contiguous(elsize, MPI_BYTE, &dtype);
     MPI_Type_commit(&dtype);
 
+    /*Set global data for the comparison*/
+    MinIDCompareThisTask = ThisTask;
```

Comment:
Let's name this `_fof_compare_Group_MinIDTask_ThisTask`, to indicate it shall only be used in that function.